pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -360,40 +361,67 @@ private ParsedTimestamp parseBackendTimestamp(String str) throws SQLException {
 
     } catch (NumberFormatException nfe) {
       throw new PSQLException(
-          GT.tr("Bad value for type timestamp/date/time: {0}", str),
+          GT.tr("Bad value for type timestamp/date/time: {0}", new String(s)),
           PSQLState.BAD_DATETIME_FORMAT, nfe);
     }
 
     return result;
   }
 
+  ParsedTimestamp parseDate(byte[]dateBytes) {
+    ParsedTimestamp parsedTimestamp = new ParsedTimestamp();
+    int length = dateBytes.length;
+
+    if (dateBytes[length - 2] == 'B' && dateBytes[length - 1] == 'C') {
+      length = length - 3;
+      parsedTimestamp.era = GregorianCalendar.BC;
+    }
+    int var1 = 0;
+    for (parsedTimestamp.year = 0; dateBytes[var1] != 45; parsedTimestamp.year = parsedTimestamp.year * 10 + (dateBytes[var1++] - 48)) {
+    }
+
+    ++var1;
+
+    for (parsedTimestamp.month = 0; dateBytes[var1] != 45; parsedTimestamp.month = parsedTimestamp.month * 10 + (dateBytes[var1++] - 48)) {
+    }
+
+    ++var1;
+
+    for (parsedTimestamp.day = 0; var1 < length; parsedTimestamp.day = parsedTimestamp.day * 10 + (dateBytes[var1++] - 48)) {
+
+    }
+
+    return parsedTimestamp;
+  }
+
   /**
    * Parse a string and return a timestamp representing its value.
    *
    * @param cal calendar to be used to parse the input string
-   * @param s The ISO formated date string to parse.
-   * @return null if s is null or a timestamp of the parsed string s.
+   * @param bytes The ISO formated date string to parse.
+   * @return null if bytes is null or a timestamp of the parsed bytes bytes.
    * @throws SQLException if there is a problem parsing s.
    */
   public @PolyNull Timestamp toTimestamp(@Nullable Calendar cal,
-      @PolyNull String s) throws SQLException {
+      byte @PolyNull []bytes) throws SQLException {
```

Comment:
`toTimestamp` like the other methods had `public` visibility, so changing the signature is not a backward-compatible change. We might keep the old methods as deprecated and implement them as `toTimestamp(inputString.getBytes(StandardCharsets.UTF_8))`